### ISSUE TITLE
Pro Form CarbonT7 Not Connecting

### ISSUE THREAD

**Issue by FurlongFX** [cagnulein/qdomyos-zwift#2474]
Attached is a copy of my latest log file hopefully demonstrating the issue.
[debug-Tue_Jul_23_12_16_48_2024.log](https://github.com/user-attachments/files/16353476/debug-Tue_Jul_23_12_16_48_2024.log)

**Comment by cagnulein** [comment]
checked.
you have to fully remove the ip setting from here `Proform IP` (be sure to don't leave any white spaces at all and press ok)
and instead set it to `Nordictrack 2950 IP`. press ok and restart qz
Let me know

**Comment by FurlongFX** [comment]
I removed the ip setting, however there is no option for Nordictrack 2950 IP. Attached is the options I'm given in the QZ app.
![treadmillOptionsScreenGrab](https://github.com/user-attachments/assets/5c608f71-fec7-406b-9f91-00eca286e53d)

**Comment by FurlongFX** [comment]
ohhh I see now, sorry. Coming right up

**Comment by FurlongFX** [comment]
[debug-Tue_Jul_23_12_36_48_2024.log](https://github.com/user-attachments/files/16353610/debug-Tue_Jul_23_12_36_48_2024.log)

**Comment by cagnulein** [comment]
ok now i can see a connection from the companion!
which device did you select now in the companion?
i will go to bed soon, in case i will answer you back tomorrow

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 23 lug 2024 alle 21:39 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
right now it's set to "other"
not sure what to set it to

**Comment by cagnulein** [comment]
try to set to x22i and restart the treadmill
let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 23 lug 2024 alle 21:44 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
[debug-Tue_Jul_23_12_51_53_2024.log](https://github.com/user-attachments/files/16353719/debug-Tue_Jul_23_12_51_53_2024.log)

**Comment by cagnulein** [comment]
it's worst, i can't see the connection anymore.
try the t7.5s instead

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 23 lug 2024 alle 21:53 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
here's the log with t7.5s

[debug-Tue_Jul_23_13_09_01_2024.log](https://github.com/user-attachments/files/16353862/debug-Tue_Jul_23_13_09_01_2024.log)

**Comment by cagnulein** [comment]
it seems that since you changed from other, i can't see any frames. that's
really strange.
are you able to get the ./wolflogs/ file from the treadmill tablet via adb
or a file manager? i would like to check them
in the meantime can you please set it back to other so at least i would
like to see if we can return to the previous condition
thanks

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 23 lug 2024 alle ore 22:10 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
[debug-Tue_Jul_23_13_20_01_2024.log](https://github.com/user-attachments/files/16353969/debug-Tue_Jul_23_13_20_01_2024.log)

**Comment by cagnulein** [comment]
yes connection is back. are you able to get the /.wolflogs/
?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 23 lug 2024 alle 22:22 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
Lemme see if I can access those

**Comment by FurlongFX** [comment]
[logcat.txt](https://github.com/user-attachments/files/16354035/logcat.txt)

**Comment by FurlongFX** [comment]
[2024-07-23_logs.0.txt](https://github.com/user-attachments/files/16354114/2024-07-23_logs.0.txt)

**Comment by cagnulein** [comment]
ok that's the reason: i don't see any metrics here in the log. Can you pull all the contents from that directory? 
i would like to check if there is any other file useful there.
another thing that we can try is to restore to the factory the fitness device, and WITHOUT updating ifit, try again, it could be that ifit removed the logging thing in the last version
let me know

**Comment by FurlongFX** [comment]
Sure I'll give you all the logs in the folder, one is empty, but there IS another one

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by FurlongFX** [comment]
I'm not sure how to get to factory settings without it automatically updating howrver

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by cagnulein** [comment]
ok thanks!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 24 lug 2024 alle ore 07:22 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
[.wolflogs.zip](https://github.com/user-attachments/files/16357638/default.wolflogs.zip)

**Comment by cagnulein** [comment]
thanks, unfortunately confirmed. no metrics at all in the log
Let me know how it goes the reset thing
Thanks

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 24 lug 2024 alle ore 07:38 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
I got it to work! I just had to redo all the steps, except once I reset to factory defaults, and connected to wifi, I blocked all the ifit update sites to the tread via my router settings. I set the companion app to x22i like you mentioned above, and once I started the QZ app, I was able to see all the metrics! Just gotta figure out how to set it to MPH instead of KmPH (dumb american lol)

**Comment by FurlongFX** [comment]
it now STAYS in priviledged mode when I restart

**Comment by cagnulein** [comment]
that's great! I told you!
you can change the unit in the general settings, miles unit!

**Comment by FurlongFX** [comment]
thank you so much for all of your help, you've been so patient and informative. One coder to another, much appreicated :)

**Comment by cagnulein** [comment]
remember to rate the app on the store, it means a lot to me! thanks!

**Comment by FurlongFX** [comment]
omg of COURSE! I saw someone else on the boards was having trouble with my exact treadmill, I think I can send him my procedure so he can enjoy it too! 

Just one more question, I'm using this for Peleton. Do I input all my peleton user info in the QZ settings in order to have Peleton control my speed and incline?

**Comment by cagnulein** [comment]
> omg of COURSE! I saw someone else on the boards was having trouble with my exact treadmill, I think I can send him my procedure so he can enjoy it too!

thanks!

> 
> Just one more question, I'm using this for Peleton. Do I input all my peleton user info in the QZ settings in order to have Peleton control my speed and incline?

yes exactly. are you already able to control speed and inclination from the qz tiles without peloton?

**Comment by cagnulein** [comment]
> omg of COURSE! I saw someone else on the boards was having trouble with my exact treadmill, I think I can send him my procedure so he can enjoy it too!

thanks!

> 
> Just one more question, I'm using this for Peleton. Do I input all my peleton user info in the QZ settings in order to have Peleton control my speed and incline?

yes exactly. are you already able to control speed and inclination from the qz tiles without peloton?

**Comment by FurlongFX** [comment]
ahhh, I'm actually NOT able to do that currently, Are there specific settings to allow that?

**Comment by cagnulein** [comment]
when you push the plus button on the speed or inclination, do you see a fake touch event on the treadmill touchscreen?

**Comment by FurlongFX** [comment]
nah, I don't.

**Comment by FurlongFX** [comment]
I think it's because the version of iFit that's currently running doesn't even HAVE the screen touch options

**Comment by FurlongFX** [comment]
I wonder if I should push a specific iFit version to the tread, one that has the touch speed options but will also remain in priviledged mode

**Comment by FurlongFX** [comment]
ahhh yeah, just figured that one out. Ok

**Comment by cagnulein** [comment]
there is a setting for this usually in ifit

**Comment by FurlongFX** [comment]
other thing I noticed is that although priviledged mode remains active upon restart, USB debugging gets shut off automatically. I have to enable it each time I restart

**Comment by FurlongFX** [comment]
I'm sending you a zip of my bug report. I'm hoping you can help me enable QZ control for speed and incline, and hopefully, I'll be done with the setup!
[btsnoop_hci.log](https://github.com/user-attachments/files/16371665/btsnoop_hci.log)

**Comment by FurlongFX** [comment]
the actions I took while it was logging was open QZ, start a manual workout, adjust speed and incline, then turn off QZ

**Comment by cagnulein** [comment]
no i don't need the hci snoof :) the process goes through wifi. did you enable the sliders now?

**Comment by FurlongFX** [comment]
yeah the sliders are enabled, but there's no control from QZ to speed or incline. And no touch responses on the iFit screen when I hit the + or - on QZ

**Comment by FurlongFX** [comment]
still getting all the readings from tread to QZ though, so that's a plus

**Comment by cagnulein** [comment]
try to enable the adb remote settings in qz in the nordictrack treadmill setting. restart qz and try again. we have to see a touch event on the tablet when you're playing with the plus and minus button on qz

**Comment by cagnulein** [comment]
ok so from your pc or mac run this

adb shell input swipe 100 200 300 400 100

you have to see a touch reaction on the treadmill

**Comment by FurlongFX** [comment]
unfortunately, I get "adb.exe: more than one device/emulator"

**Comment by cagnulein** [comment]
directly from chatgpt

```
The error message "adb.exe: more than one device/emulator" indicates that the Android Debug Bridge (ADB) is detecting multiple devices or emulators connected to your computer, and it doesn't know which one to target for your command.

To resolve this issue, you can specify which device or emulator you want to communicate with by using the device ID. Here’s how you can do it:

1. **List All Devices:**
   Open a command prompt or terminal and enter the following command to list all connected devices and emulators:

   ```bash
   adb devices
   ```

   This will return a list of all devices and their IDs. It might look something like this:

   ```
   List of devices attached
   emulator-5554   device
   emulator-5556   device
   1234567abcdef   device
   ```

2. **Specify the Device:**
   Once you have the device ID you want to use, you can specify it in your ADB command using the `-s` flag. For example, if you want to target `emulator-5554`, you would use:

   ```bash
   adb -s emulator-5554 <command>
   ```

   Replace `<command>` with the actual command you want to execute, such as `install`, `shell`, etc.

3. **Disconnect Unneeded Devices:**
   If possible, disconnect any devices or emulators that are not needed to reduce confusion.

4. **Use `adb -e` or `adb -d` Flags:**
   - Use `adb -e <command>` if you want to target the emulator.
   - Use `adb -d <command>` if you want to target a physical device.

   These flags are helpful if you only have one emulator or one device connected.

By using these methods, you should be able to direct your ADB commands to the correct device or emulator. Let me know if you need further assistance!
```

**Comment by cagnulein** [comment]
ok so try now the swipe things

**Comment by FurlongFX** [comment]
ok great, when I inputted a swipe from 200 0 to 200 100, I saw the bottom panel get brought up

**Comment by FurlongFX** [comment]
I'm just trying to map out the coordinates of my exact screen

**Comment by FurlongFX** [comment]
success! This was a swipe 75 430 75 306
https://github.com/user-attachments/assets/caeddfb8-7ffc-4ff4-b30a-976d7cfc0e15

**Comment by FurlongFX** [comment]
![Thu07-25-2024@11-06-35 17](https://github.com/user-attachments/assets/f6f666b2-33ff-49d9-967f-9cb88771b1c0)
 this is a screenshot taken from the adb-screenshot.bat

**Comment by FurlongFX** [comment]
the x coords of the incline and speed, respectively, are 75 and 950
the y coords from bottom to top of the sliders are 430(btm) to 185(top)

**Comment by cagnulein** [comment]
ok perfect, so now are you able, pressing the plus button on the speed or incline from qz, to see an event there?

**Comment by FurlongFX** [comment]
no I'm not. I'm thinking it's because whichever model is being assigned by QZ doesn't align with the coordinates of my model's screen

**Comment by FurlongFX** [comment]
the Carbon T7 is only a 7" touch screen, and the companion app is assuming an x22i which is a 22" screen

**Comment by FurlongFX** [comment]
the T7.5s you mentioned above has the exact same screen size as mine. Should I try that one instead on the companion app?

**Comment by cagnulein** [comment]
no that's not the reason, qz is sending valid coordinates when you push the plus, maybe not perfectly calibrated but you should see an event. maybe it's because you are still connected with the pc via adb? only one adb connection is allowed to the treadmill.

the companion app, when you enabled the adb remote (as we want) it doesn't consider the device selected there

**Comment by FurlongFX** [comment]
ahh ok
could this be why? I'm seeing this extra "entity" on ADB explorer

![image](https://github.com/user-attachments/assets/ce6147e5-afa7-4416-8cb6-3788a8b7e0eb)

**Comment by cagnulein** [comment]
could be...but qz is connecting directly to the ip that you set in the settings, so it should avoid the emulator. try to send me a fresh debug log of qz, so we will check if there is any adb error

**Comment by FurlongFX** [comment]
[debug-Thu_Jul_25_11_39_21_2024.log](https://github.com/user-attachments/files/16382461/debug-Thu_Jul_25_11_39_21_2024.log)

**Comment by cagnulein** [comment]
unfortunately the log is truncated from the start. close qz before sending
the log, otherwise it will be corrupted
thanks

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 25 lug 2024 alle ore 20:46 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
ahh sorry, I didn't realize it continues to write the log after the app is closed. It's still writing it, I'll send it once it's finished

**Comment by FurlongFX** [comment]
[debug-Thu_Jul_25_12_02_43_2024.log](https://github.com/user-attachments/files/16382641/debug-Thu_Jul_25_12_02_43_2024.log)

**Comment by cagnulein** [comment]
i don't see any error: are you able to connect your phone to the pc and run
an adb logcat of the phone while qz is starting? I would like to see if
there is any underlying errors
Let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 25 lug 2024 alle ore 21:04 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
[logcat.txt](https://github.com/user-attachments/files/16383173/logcat.txt)

**Comment by cagnulein** [comment]
is it possibile that you stripped away something? i can't see any reference
of the java module AdbRemote

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 25 lug 2024 alle ore 22:01 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
I confirmed, adb remote is active on QZ. I cleared the log, then ran it again after having opened the QZ app and pasted it into a txt file. I just used the standard "adb logcat - d" command

**Comment by cagnulein** [comment]
ok 10pm here. i guess i have to add some debug log to understand what's
going on. because it simply that qz doesn't start the adb thing.
I will let you know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 25 lug 2024 alle ore 22:11 FurlongFX ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
ok i added some debug logs, give it a try when it finished to build it
https://github.com/cagnulein/qdomyos-zwift/actions/runs/10110231384

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 25 lug 2024 alle ore 22:13 cagnulein ***@***.***> ha
scritto:

**Comment by FurlongFX** [comment]
[logcat_v3.txt](https://github.com/user-attachments/files/16396611/logcat_v3.txt)

**Comment by FurlongFX** [comment]
[logcat4.txt](https://github.com/user-attachments/files/16396684/logcat4.txt)

this one is from my Pixel 8 Pro, the previous one is using my Pixel 6

**Comment by FurlongFX** [comment]
I feel like this is the culprit
![image](https://github.com/user-attachments/assets/c5ef97f4-86c2-4c93-bffe-03f989c04d89)

**Comment by cagnulein** [comment]
for sure that's the error. Strange thing I can't see my new logs, did you install the new build? Anyway yes, there is a connection refused so the possibility are:
1) the ip is incorrect, i'm excluding this
2) the port is incorrect: which is the port when you connect from the pc?
3) there are other adb instances connected to your treadmill, close them all
4) this treadmill have some strange security policy that inhibits a connection from another android device

Let me know your thoughts

**Comment by FurlongFX** [comment]
IP should be fine. Port is 5555. Shouldn't be any other adb connections. Any way to check that? 

As far as the latest build, I downloaded the latest APK and tried installing it from my files folder on my phone, but it errored out. Do I have to install it via ADB?

**Comment by cagnulein** [comment]
you have to remove first the original qz , and install the apk.

anyway we already saw the error.

the only way to be sure about no other connections is to reboot the
treadmill and connect with QZ Right after.

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 27 lug 2024 alle 17:36 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
so I restarted the tread and only connected my pixel 6. I'm using that one as my QZ device, and my current phone is being used as my peleton device. 

Attached is a log using the latest nightly build.
[logcatnew.txt](https://github.com/user-attachments/files/16400974/logcatnew.txt)

**Comment by FurlongFX** [comment]
okay I think we have success here

![image](https://github.com/user-attachments/assets/9c061f62-1dde-4e35-abde-f4b83db30bff)

**Comment by cagnulein** [comment]
yes it seems so, what did you change? do you see now the event on the treadmill?

**Comment by FurlongFX** [comment]
so the one thing that I noticed is that every time I restart the tread, USB debugging flips off. I have to remember to turn it on every time. Can that be included in the QZ startup sequence? 

As far as seeing the event, I can't. I'm wondering if the swipe up coords are out of the bounds of my screen. Because when you sent me the shell swipe command to test it earlier, it didn't work, but DID when I adjusted the command coordinates

**Comment by FurlongFX** [comment]
the x coordinate on the log are way past the max x coord on my screen
![image](https://github.com/user-attachments/assets/e7269ec2-3da2-44e9-9687-abb7342b4287)

**Comment by cagnulein** [comment]
no, it can't be included in the startup sequence because without that one qz can't communicate at all with the treadmill.

OK, I will let the profile of your treadmill with the new coordinates in the next days

**Comment by cagnulein** [comment]
> the x coordinate on the log are way past the max x coord on my screen ![image](https://private-user-images.githubusercontent.com/176003436/352769438-e7269ec2-3da2-44e9-9687-abb7342b4287.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIxMDI4NjMsIm5iZiI6MTcyMjEwMjU2MywicGF0aCI6Ii8xNzYwMDM0MzYvMzUyNzY5NDM4LWU3MjY5ZWMyLTNkYTItNDRlOS05Njg3LWFiYjczNDJiNDI4Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyN1QxNzQ5MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YzZlNDM3NjY4NTFkMDhjMTczMGYxY2I2ZWJjYzA1YWQ0M2RlMmJiZDNjYmQ1ZTAyOGIwZGI3N2JiODkzN2U3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.lQ6U-h_XT8f_rSdM9jbbo5ZPK3Rw0Wr24GQW9DlnMe0)

the 75 should be fine

**Comment by FurlongFX** [comment]
what I'm saying is, the tread disables it on every startup. What I'm asking is if the qz companion startup sequence can force USB-debugging ON

**Comment by cagnulein** [comment]
no it can't because it would need root access.
or at least it's something that i never tried. if you want we can
experiment on this.

are you able to collect some information about allowing root permissions to
an apk and the java commands to enable it ?

also, I'm wondering if there is already some app like tasker, That can do
these already

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 27 lug 2024 alle 19:50 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
not sure if this pertains to what you're looking for
it includes a java method for checking then enabling

https://www.reddit.com/r/tasker/comments/5k5o25/how_to_enable_usb_debugging/

**Comment by cagnulein** [comment]
i will investigate this. let's first check the coordinates thing

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 27 lug 2024 alle 19:57 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
oh of course, thanks so much :)

**Comment by FurlongFX** [comment]
does the treadmill option in the app modify the coordinates in the shell command?

**Comment by cagnulein** [comment]
for the inclination, x, are basically all the same to 75

anyway you can try the profiles

x22i or t8.5s or x14i from the combo box on the qz settings nordictrack
treadmill setting model

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 27 lug 2024 alle 20:49 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
yeah, the profiles all send correct metrics, but no swipe response from app

**Comment by cagnulein** [comment]
ok i will add the profiles for yours tomorrow, let's see if it makes any
difference

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 28 lug 2024 alle ore 22:42 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
copy that, thanks so much :)

**Comment by FurlongFX** [comment]
if it works, there's many people on the facebook group with my exact model that I can share my procedure with.

**Comment by cagnulein** [comment]
@FurlongFX done

https://github.com/cagnulein/qdomyos-zwift/actions/runs/10139788036

enable the treadmill model in the treadmill combobox and restart qz

let me know

**Comment by FurlongFX** [comment]
I installed the latest nightly build from 18 hrs ago, and I don't see my model in "Specific Model" dropdown list

**Comment by cagnulein** [comment]
it's the last one in the list and it was from 11 hours ago
https://github.com/cagnulein/qdomyos-zwift/actions/runs/10139788036


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 29 lug 2024 alle 21:01 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
okay, so got it installed and good news, it KINDA worked! However, hitting the plus sign jumped it up like 9 full pts on each metric respectively

**Comment by FurlongFX** [comment]
[logcat.txt](https://github.com/user-attachments/files/16418140/logcat.txt)

**Comment by cagnulein** [comment]
ok please create a debug log with the issue. thanks

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 29 lug 2024 alle ore 21:20 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
[debug-Mon_Jul_29_12_22_55_2024.log](https://github.com/user-attachments/files/16418174/debug-Mon_Jul_29_12_22_55_2024.log)

**Comment by FurlongFX** [comment]
ignore that one
[debug-Mon_Jul_29_12_22_55_2024.log](https://github.com/user-attachments/files/16418187/debug-Mon_Jul_29_12_22_55_2024.log)

**Comment by cagnulein** [comment]
fixed https://github.com/cagnulein/qdomyos-zwift/actions/runs/10150584136

**Comment by FurlongFX** [comment]
awesome! I'll try it out once it finishes! 
One more thing, is installing the app this way putting me into a trial-mode? I seem to be getting a timeout pushing me to your patreon

**Comment by cagnulein** [comment]
yes i will unlock it if it works :)

**Comment by FurlongFX** [comment]
alright alright! Super close! the incline goes up .5 with every "+" hit, so that's perfect. The speed goes up .3 with every "+" hit. I expect it should be limited to .1 increases. So VERY VERY close

**Comment by cagnulein** [comment]
that's perfect! it's going .3 because by default the speed step in the treadmill setting is .5 km/h converted to .3 mi/h

you can try to set to .1 but due to the fact that qz is handling all in km/h on the lower layer could be a floating issue. anyway, this not mean that it can't handle all the decimal, it's just a plus minus button issue. in a peloton workout everything should works smoothly

let me know

**Comment by FurlongFX** [comment]
oh man, awesome! I have a 6 mile peloton run today, so I'll boot that bad boy up!

**Comment by cagnulein** [comment]
Awesome, good luck!
Going to bed now, have a great run!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 29 lug 2024 alle ore 23:03 FurlongFX ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
also i suggest to try it with a test run first, even without you on the treadmill :)

**Comment by FurlongFX** [comment]
thank you Sir! Will do!

**Comment by FurlongFX** [comment]
so everything seems to be working except peloton. Peloton can auto set incline correctly, but doesn't set speed. And after the incline gets autoset, I can't set the speed from QZ either, it seems to tap the empty area

**Comment by cagnulein** [comment]
create a debug log and i will check it
8pm here and i'm going to run. i will check it tomorrow
thanks

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 30 lug 2024 alle 19:58 FurlongFX ***@***.***>
ha scritto:

**Comment by FurlongFX** [comment]
of course, will do. Have a good run Sir

**Comment by cagnulein** [comment]
any news @FurlongFX  ?

**Comment by FurlongFX** [comment]
yeah, sorry dude! I just started a new freelance gig, and I've been super busy, I'll upload that log today

**Comment by FurlongFX** [comment]
[debug-Fri_Aug_2_15_52_07_2024.log](https://github.com/user-attachments/files/16475960/debug-Fri_Aug_2_15_52_07_2024.log)
here's the latest log. 

Peloton connects, and adjusts INCLINE half-way within the min and max recommended. Speed, however, does NOT adjust.

**Comment by cagnulein** [comment]
For the inclination you have to select the right difficulty in the peloton
settings in qz

For the speed you have to enable the speed force setting in the treadmill
settings in qz. Disabled by default to safety reasons

Let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 3 ago 2024 alle 01:00 FurlongFX ***@***.***> ha
scritto:

**Comment by FurlongFX** [comment]
all right! Everything's working! Thanks again for your continued support. So for now, how do I set my current build as the paid version?

**Comment by cagnulein** [comment]
Send me an email to Roberto.viola83 at Gmail.com indicating this ticket and I will unlock it 
Also you were mentioning about a Facebook group, could you please link it here?
